### PR TITLE
CORDA-3096 Formatted database migration logs - docs

### PR DESCRIPTION
**Pull request by szymonsztuka**
Formatted database migration logs - documentation.
Indirectly depends on https://github.com/corda/corda/pull/5307.

### CODE REVIEW

File: docs/source/node-database-initialisation-logging.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ [ START ]
+ [ CHANGE_SETS_COUNT ]
+ [ CHANGE_SETS ]
+ [ PROGRESS ]
+ [ END | ERROR ]
+
+where:
+
+* START:= a log line denoting the start of database schema setup or update
+* CHANGE_SETS_COUNT:=number of change sets detected as needed to be applied against the database
```
Comment (gwr3com) [inline docs/source/node-database-initialisation-logging.rst]: Some items have a space after ":=".  This line doesn't.

File: docs/source/node-database-initialisation-logging.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+.. sourcecode:: none
+
+ [ START ]
+ [ CHANGE_SETS_COUNT ]
+ [ CHANGE_SETS ]
+ [ PROGRESS ]
+ [ END | ERROR ]
+
+where:
```
Comment (gwr3com) [inline docs/source/node-database-initialisation-logging.rst]: A small inconsistency: "where" here vs. "Where" elsewhere in this file

File: docs/source/node-database-initialisation-logging.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+The log line format:
+
+.. sourcecode:: none
+
+    databaseInitialisation(id=<RANDOM_ID>;[changeset=<ID>];status=<STATUS>[;error_code=<CODE>;message=<ERROR>])
+
+Where:
+
+* RANDOM_ID:= random generated value allowing to grep all logs form withing a single migration run
```
Comment (gwr3com) [inline docs/source/node-database-initialisation-logging.rst]: form -> from

File: docs/source/node-database-initialisation-logging.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+* CHANGE_SET_ID:= Liquibase change set ID
```
Comment (gwr3com) [inline docs/source/node-database-initialisation-logging.rst]: CHANGE_SET_ID here vs. ID above

File: docs/source/node-database-initialisation-logging.rst [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+* STATUS:= the status for the entire database schema initialisation process and a specific change sets:
+  |        "start" - start og the whole database initialisation process,
+  |        "to be run" - the list of change sets before migration run,
+  |        "started" - the change set is about to be run,
+  |        "successful" - the change sets has been run, "Error" - an error occurred when running the change set
+  |        "error" - an error for the whole process or an error while running a specific changeset set
+*  ERROR:= a detailed message, for change set error it will be error produced by Liquibase
```
Comment (gwr3com) [inline docs/source/node-database-initialisation-logging.rst]: Should CODE be explained before ERROR ?

File: docs/source/node-database-initialisation-logging.rst [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    databaseInitialisation(id=FrSzFgm2;status="error";message="Migration failed for change set migration/node-services.changelog-init.xml::1511451595465-39::R3.Corda:      Reason: liquibase.exception.DatabaseException: Table "NODE_MESSAGE_RETRY" not found; SQL statement: ALTER TABLE PUBLIC.node_message_retry ADD CONSTRAINT node_message_retry_pkey PRIMARY KEY (message_id) [42102-197] [Failed SQL: ALTER TABLE PUBLIC.node_message_retry ADD CONSTRAINT node_message_retry_pkey PRIMARY KEY (message_id)]")
+
+Detailed logs
+^^^^^^^^^^^^^
+
+The native Liqubase logs are disabled by default.
+They can be enabled by adding an extra log4j2 file with 'INFO' log level for 'liqubase' package:
```
Comment (gwr3com) [inline docs/source/node-database-initialisation-logging.rst]: 'liqubase' or 'liquibase' ?

**Comment by szymonsztuka** [comment]
All @gwr3com comments addressed.

File: docs/source/node-database-initialisation-logging.rst [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    databaseInitialisation(id=FrSzFgm2;changeset="migration/common.changelog-init.xml::1511451595465-1.1::R3.Corda";status="error";message="Migration failed for change set migration/node-services.changelog-init.xml::1511451595465-39::R3.Corda:      Reason: liquibase.exception.DatabaseException: Table "NODE_MESSAGE_RETRY" not found; SQL statement: ALTER TABLE PUBLIC.node_message_retry ADD CONSTRAINT node_message_retry_pkey PRIMARY KEY (message_id) [42102-197] [Failed SQL: ALTER TABLE PUBLIC.node_message_retry ADD CONSTRAINT node_message_retry_pkey PRIMARY KEY (message_id)]")
+    databaseInitialisation(id=FrSzFgm2;status="error";message="Migration failed for change set migration/node-services.changelog-init.xml::1511451595465-39::R3.Corda:      Reason: liquibase.exception.DatabaseException: Table "NODE_MESSAGE_RETRY" not found; SQL statement: ALTER TABLE PUBLIC.node_message_retry ADD CONSTRAINT node_message_retry_pkey PRIMARY KEY (message_id) [42102-197] [Failed SQL: ALTER TABLE PUBLIC.node_message_retry ADD CONSTRAINT node_message_retry_pkey PRIMARY KEY (message_id)]")
+
+Detailed logs
+^^^^^^^^^^^^^
+
+The native Liquibase logs are disabled by default.
```
Comment (anthonykeenan) [inline docs/source/node-database-initialisation-logging.rst]: Native Liquibase logs....

File: docs/source/node-database-initialisation-logging.rst [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+They can be enabled by adding an extra log4j2 file with 'INFO' log level for 'liquibase' package:
```
Comment (anthonykeenan) [inline docs/source/node-database-initialisation-logging.rst]: for the `liquibase` logger

File: docs/source/node-database-initialisation-logging.rst [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    databaseInitialisation(id=FrSzFgm2;change_set_count="2")
+    databaseInitialisation(id=FrSzFgm2;changeset="migration/common.changelog-init.xml::1511451595465-1.1::R3.Corda";status="to be run")
+    databaseInitialisation(id=FrSzFgm2;changeset="migration/vault-schema.changelog-init.xml::1511451595465-22::R3.Corda";status="to be run")
+    databaseInitialisation(id=FrSzFgm2;changeset="migration/common.changelog-init.xml::1511451595465-1.1::R3.Corda";status="started")
+    databaseInitialisation(id=FrSzFgm2;changeset="migration/common.changelog-init.xml::1511451595465-1.1::R3.Corda";status="successful")
+    databaseInitialisation(id=FrSzFgm2;changeset="migration/vault-schema.changelog-init.xml::1511451595465-22::R3.Corda";status="started")
+    databaseInitialisation(id=FrSzFgm2;changeset="migration/vault-schema.changelog-init.xml::1511451595465-22::R3.Corda";status="successful")
+    databaseInitialisation(id=FrSzFgm2;status="successful")
+
+The example unsuccessful database initialisation logs:
```
Comment (anthonykeenan) [inline docs/source/node-database-initialisation-logging.rst]: An...... log:

File: docs/source/node-database-initialisation-logging.rst [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+* STATUS := the status for the entire database schema initialisation process and a specific change sets:
+  |        "start" - start og the whole database initialisation process,
+  |        "to be run" - the list of change sets before migration run,
+  |        "started" - the change set is about to be run,
+  |        "successful" - the change sets has been run,
+  |        "error" - an error for the whole process or an error while running a specific change set
+*  CODE := a predefined error code, the list will be published later
+*  ERROR := a detailed message, for change set error it will be error produced by Liquibase
+
+The example database initialisation logs:
```
Comment (anthonykeenan) [inline docs/source/node-database-initialisation-logging.rst]: An ... log:

File: docs/source/node-database-initialisation-logging.rst [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+.. sourcecode:: none
+
+    databaseInitialisation(id=<RANDOM_ID>;[changeset=<ID>];status=<STATUS>[;error_code=<CODE>;message=<ERROR>])
+
+where:
+
+* RANDOM_ID := random generated value allowing to grep all logs from withing a single migration run
+* ID := Liquibase change set ID
+* STATUS := the status for the entire database schema initialisation process and a specific change sets:
+  |        "start" - start og the whole database initialisation process,
```
Comment (anthonykeenan) [inline docs/source/node-database-initialisation-logging.rst]: of .... database migration process

File: docs/source/node-database-initialisation-logging.rst [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+  |        "to be run" - the list of change sets before migration run,
```
Comment (anthonykeenan) [inline docs/source/node-database-initialisation-logging.rst]: the list of change sets to run

File: docs/source/node-database-initialisation-logging.rst [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+  |        "started" - the change set is about to be run,
```
Comment (anthonykeenan) [inline docs/source/node-database-initialisation-logging.rst]: the start of the current change set

File: docs/source/node-database-initialisation-logging.rst [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+  |        "successful" - the change sets has been run,
```
Comment (anthonykeenan) [inline docs/source/node-database-initialisation-logging.rst]: the successful completion of a change set

File: docs/source/node-database.rst [hunk 15]
```diff
@@ -152,3 +152,10 @@ For example:
         maximumPoolSize = 10
         connectionTimeout = 50000
     }
+
+
+Database Schema Initialisation Logging
```
Comment (anthonykeenan) [inline docs/source/node-database.rst]: Initialisation=Migration

File: docs/source/node-database-initialisation-logging.rst [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+ [ END | ERROR ]
+
+where:
+
+* START := a log line denoting the start of database schema setup or update
+* CHANGE_SETS_COUNT := number of change sets detected as needed to be applied against the database
+* CHANGE_SETS := a list of all migrations to run, specified one by log line
+* PROGRESS := log lines before and after running each change set
+* END := a line denoting the successful end of database initialisation
+* ERROR := any error related to a database connection and setup e.g. the node database misconfiguration preventing the node to start, or an error related to creating database schema, a list of predefined error codes will be documented
```
Comment (anthonykeenan) [inline docs/source/node-database-initialisation-logging.rst]: I think we can just replace this with 'information about errors that occurred during migration, if any'

File: docs/source/node-database.rst [hunk 17]
```diff
@@ -152,3 +152,10 @@ For example:
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Database initialisation and upgrade logs for Corda internal tables follows a structured format
```
Comment (anthonykeenan) [inline docs/source/node-database.rst]: Database migration logs for.... follow...

File: docs/source/node-database-initialisation-logging.rst [hunk 18]
```diff
@@ -0,0 +1,104 @@
+.. _database-initialisation-logging:
+
+Database Schema Initialisation Logging
+======================================
+
+Database initialisation and upgrade for Corda internal tables (and other objects like sequences)
```
[the remark this answers is not in the document]
Comment (anthonykeenan) [reply docs/source/node-database-initialisation-logging.rst]: Not sure I've got my apostrophe's right there... @edward-prosser might have a better idea if he has time to take a quick look

File: docs/source/node-database-initialisation-logging.rst [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+* CHANGE_SETS := a list of all migrations to run, specified one by log line
+* PROGRESS := log lines before and after running each change set
+* END := a line denoting the successful end of database initialisation
+* ERROR := any error related to a database connection and setup e.g. the node database misconfiguration preventing the node to start, or an error related to creating database schema, a list of predefined error codes will be documented
+
+The log line format:
+
+.. sourcecode:: none
+
+    databaseInitialisation(id=<RANDOM_ID>;[changeset=<ID>];status=<STATUS>[;error_code=<CODE>;message=<ERROR>])
```
Comment (anthonykeenan) [inline docs/source/node-database-initialisation-logging.rst]: Just anothe minor point - in @bpaunescu's PR the log 'events' start with a capital letter - can we make sure the two things are consistent please?

File: docs/source/node-database-initialisation-logging.rst [hunk 20]
```diff
@@ -0,0 +1,104 @@
+.. _database-initialisation-logging:
+
+Database Schema Initialisation Logging
```
[the remark this answers is not in the document]
Comment (szymonsztuka) [reply docs/source/node-database-initialisation-logging.rst]: I don't like "Migration" as it is misleading,  on the Liquibase page there is no mention of "migration" as well. The proble is I'm starting a fresh Corda node, why do I have migrate database?

File: docs/source/node-database-initialisation-logging.rst (hunk 19 above)
Comment (bpaunescu) [reply docs/source/node-database-initialisation-logging.rst]: Not sure it matters, at least not now. This doesn't print to the detailed log file.

Comment (anthonykeenan) [reply docs/source/node-database-initialisation-logging.rst]: I'd like to change it anyway while we can!

**Comment by anthonykeenan** [comment]
Also can't add ENT tickets to OS commits, we'll need to create an OS one for this work and reference.

Otherwise I'm happy to merge - the comment about quotes around the ID I'm happy for us to change in the next PR

**Comment by szymonsztuka** [comment]
> Also can't add ENT tickets to OS commits, we'll need to create an OS one for this work and reference.
> 
> Otherwise I'm happy to merge - the comment about quotes around the ID I'm happy for us to change in the next PR

Added: https://r3-cev.atlassian.net/browse/CORDA-3096

**Pull request merged**